srcts/src/components/sidebar.ts

```diff
@@ -28,20 +139,96 @@ class Sidebar {
     TRANSITIONING: "transitioning",
   };
 
-  public static initCollapsible(container: HTMLElement): void {
-    // Signal that this layout is initialized by removing the init attribute
-    container.removeAttribute("data-bslib-sidebar-init");
+  /**
+   * If sidebars are initialized before the DOM is ready, we re-schedule the
+   * initialization to occur on DOMContentLoaded.
+   * @private
+   * @static
+   * @type {boolean}
+   */
+  private static onReadyScheduled = false;
+  /**
+   * A map of initialized sidebars to their respective Sidebar instances.
+   * @private
+   * @static
+   * @type {WeakMap<HTMLElement, Sidebar>}
+   */
+  private static instanceMap: WeakMap<HTMLElement, Sidebar> = new WeakMap();
+
+  /**
+   * Given a sidebar container, return the Sidebar instance associated with it.
+   * @public
+   * @static
+   * @param {HTMLElement} el
+   * @returns {(Sidebar | undefined)}
+   */
+  public static getInstance(el: HTMLElement): Sidebar | undefined {
+    return Sidebar.instanceMap.get(el);
+  }
+
+  /**
+   * Initialize all collapsible sidebars on the page.
+   * @public
+   * @static
+   */
+  public static initCollapsibleAll(): void {
+    if (document.readyState === "loading") {
+      if (!Sidebar.onReadyScheduled) {
+        Sidebar.onReadyScheduled = true;
+        document.addEventListener("DOMContentLoaded", () => {
+          Sidebar.initCollapsibleAll();
+        });
+      }
+      return;
+    }
```

Comment:
Seems like a web-component approach could get rid of this (and some other init) code (as well as the inline `<script>`). I'm OK with keeping this as is (for the sake of moving on to other things), but maybe worth looking into if you want an excuse to learn more about web components.

Reply:
Yeah web components seems promising (for another day)!